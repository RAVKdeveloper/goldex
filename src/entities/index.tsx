export { TitlePage } from './TitlePage/ui/TitlePage.entity'
export { Balances } from './Balances/ui/Balances.entity'
export { AddingModalEntity } from './Modals/Adding/AddingModal.entity'
export { WithdrawModalEntity } from './Modals/Withdraw/ui/WithdrawModal.entity'
export { NotFoundInvoicesEntity } from './NotFounds/Invoices/ui/NotFoundInvoices.entity'
