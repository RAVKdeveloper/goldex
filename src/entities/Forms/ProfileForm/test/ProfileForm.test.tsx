import { render, screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'

import { ProfileFormEntity } from '@/entities'

describe('testing ProfileFormEntity component', () => {
  const user = userEvent.setup()

  it('render and typing', async () => {
    const component = render(<ProfileFormEntity />)

    const rootComponent = screen.getByTestId('profileForm')
    const uploadAvatarEl = screen.getByTestId('uploadAvatar')
    const textareComponent = screen.getByTestId('input')

    await user.click(uploadAvatarEl)
    await user.type(textareComponent, 'input')

    expect(rootComponent).toBeInTheDocument()
    expect(component).toMatchSnapshot()
  })
})
