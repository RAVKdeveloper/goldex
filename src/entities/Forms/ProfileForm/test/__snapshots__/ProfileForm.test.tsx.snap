// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ProfileFormEntity component render and typing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="form"
        data-testid="profileForm"
      >
        <div
          class="header"
        >
          <h2
            class="title"
          >
            profileTitle
          </h2>
          <p
            class="supportingText"
          >
            profileSupText
          </p>
        </div>
        <div
          class="content"
        >
          <uploadavatar
            data-testid="uploadAvatar"
          />
          <textfield
            data-testid="input"
            label="labels.username"
            name="username"
            sizetype="lg"
          />
        </div>
        <div
          class="button"
        >
          <button
            disabled=""
            variant="primary"
          >
            btn.save
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="form"
      data-testid="profileForm"
    >
      <div
        class="header"
      >
        <h2
          class="title"
        >
          profileTitle
        </h2>
        <p
          class="supportingText"
        >
          profileSupText
        </p>
      </div>
      <div
        class="content"
      >
        <uploadavatar
          data-testid="uploadAvatar"
        />
        <textfield
          data-testid="input"
          label="labels.username"
          name="username"
          sizetype="lg"
        />
      </div>
      <div
        class="button"
      >
        <button
          disabled=""
          variant="primary"
        >
          btn.save
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
